runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -819,27 +849,32 @@ ClientSessionData::getClassRecord(ClientSessionData::ClassInfo &classInfo)
 }
 
 const AOTCacheClassRecord *
-ClientSessionData::getClassRecord(J9Class *clazz, bool &missingLoaderRecord)
+ClientSessionData::getClassRecord(J9Class *clazz, bool &missingLoaderRecord, bool &uncachedClass)
    {
    TR_ASSERT(getROMMapMonitor()->owned_by_self(), "Must hold ROMMapMonitor");
 
    auto it = getROMClassMap().find(clazz);
    if (it == getROMClassMap().end())
+      {
+      uncachedClass = true;
       return NULL;
+      }
 
-   auto record = getClassRecord(it->second);
+   bool missingLoaderInfo = false;
+   auto record = getClassRecord(it->second, missingLoaderInfo);
    missingLoaderRecord = !record;
    return record;
    }
 
 const AOTCacheClassRecord *
-ClientSessionData::getClassRecord(J9Class *clazz, JITServer::ServerStream *stream)
+ClientSessionData::getClassRecord(J9Class *clazz, JITServer::ServerStream *stream, bool &missingLoaderInfo)
    {
    const AOTCacheClassRecord *record = NULL;
    bool missingLoaderRecord = false;
+   bool uncachedClass = false;
       {
       OMR::CriticalSection cs(getROMMapMonitor());
-      record = getClassRecord(clazz, missingLoaderRecord);
+      record = getClassRecord(clazz, missingLoaderRecord, uncachedClass);
       }
 
```

Comment:
If the `getClassRecord()` call above failed because we reached the cache size limit, it makes more sense to fail immediately so that we don't communicate with the client unnecessarily.

Reply:
Isn't that what happens already? If `getClassRecord()` fails because the cache size limit was reached then both `missingLoaderInfo` and `uncachedClass` will be false.

Reply:
Right, I missed that. Maybe you should add a comment.